packages/core/core/src/factories/useUserBillingFactory.ts

```diff
@@ -0,0 +1,120 @@
+import { Ref, ComputedRef, computed } from '@vue/composition-api';
+import { UseUserBilling } from '../types';
+import { sharedRef } from '../utils';
+
+export interface UseUserBillingFactoryParams<ADDRESS> {
+  addAddress: (params: {
+    address: ADDRESS;
+    addresses: ComputedRef<ADDRESS[]>;
+  }) => Promise<ADDRESS[]>;
+  deleteAddress: (params: {
+    address: ADDRESS;
+    addresses: ComputedRef<ADDRESS[]>;
+  }) => Promise<ADDRESS[]>;
+  updateAddress: (params: {
+    address: ADDRESS;
+    addresses: ComputedRef<ADDRESS[]>;
+  }) => Promise<ADDRESS[]>;
+  load: (params: {
+    addresses: ComputedRef<ADDRESS[]>;
+  }) => Promise<ADDRESS[]>;
+  setDefault: (params: {
```

Comment:
we will need current state here as well (so value of `defaultAddress`)

Reply:
Also added to updateAddress and deleteAddress based on our conversation